### DISCUSSION TITLE
Require correct source code for [@types/jsoneditor]

### DISCUSSION THREAD

**Discussion by murray-lang** [DefinitelyTyped/DefinitelyTyped#63611] [Issues with a @types package]
This has been discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57203, with essentially a disinterested acknowledgement.  I am trying to raise this as an issue in the hope that it will be taken seriously, because the problem it presents will never go away. I instead appear to have been directed back to discussions, but anyway, here goes...

I work for a large corporation that requires access to the source code of all open source modules for the purpose of scanning for vulnerabilities. For nearly every npm module we use, the repository linked in npmjs.com contains tags or branch names that correspond to the versions we have. A few don't, including @types/jsoneditor@9.5.1. I understand that it is just types and not executable code, but I am not going to win this argument with a tick box in a large corporation vulnerable to cyber attacks.

If I cannot find the exact sources then I will be forced to uninstall @types/jsoneditor and pepper my code with "any". This is not hard, but it's a shame that DefinitelyTyped is forbidden.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
> I work for a large corporation that requires access to the source code of all open source modules for the purpose of scanning for vulnerabilities. For nearly every npm module we use, the repository linked in npmjs.com contains tags or branch names that correspond to the versions we have.

You shouldn't use code from github for scanning for vulnerabilities. The code in repository can be totally different from code which installed via npm. You must not trust.

You should scan for vulnerabilities code which you really installed - the code from npm.

**Comment by murray-lang** [comment]
Thanks Dmitry,You know this, I know this, the checkbox doesn't The checkbox wins.Get Outlook for Android
